src/pages/workspaces/workspace/analysis/modals/compute-modal-components.test.ts

```diff
@@ -0,0 +1,155 @@
+import '@testing-library/jest-dom'
+
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { h } from 'react-hyperscript-helpers'
+import {
+  PersistentDiskSection,
+  PersistentDiskType,
+} from 'src/pages/workspaces/workspace/analysis/modals/compute-modal-components'
+import {
+  defaultAutopauseThreshold,
+  defaultComputeRegion,
+  defaultComputeZone,
+  defaultDataprocMachineType,
+  defaultDataprocMasterDiskSize,
+  defaultDataprocWorkerDiskSize,
+  defaultGcePersistentDiskSize,
+  defaultGpuType,
+  defaultNumDataprocPreemptibleWorkers,
+  defaultNumDataprocWorkers,
+  defaultNumGpus,
+  defaultPersistentDiskType,
+  getDefaultMachineType,
+} from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import { toolLabels } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+
+
+const defaultIComputeConfig = {
+  selectedPersistentDiskSize: defaultGcePersistentDiskSize,
+  selectedPersistentDiskType: defaultPersistentDiskType,
+  masterMachineType: getDefaultMachineType(false, toolLabels.RStudio),
+  masterDiskSize: defaultDataprocMasterDiskSize,
+  numberOfWorkers: defaultNumDataprocWorkers,
+  numberOfPreemptibleWorkers: defaultNumDataprocPreemptibleWorkers,
+  workerMachineType: defaultDataprocMachineType,
+  workerDiskSize: defaultDataprocWorkerDiskSize,
+  componentGatewayEnabled: true,
+  gpuEnabled: false,
+  hasGpu: false,
+  gpuType: defaultGpuType,
+  numGpus: defaultNumGpus,
+  autopauseThreshold: defaultAutopauseThreshold,
+  computeRegion: defaultComputeRegion,
+  computeZone: defaultComputeZone,
+}
+
+const updateComputeConfig = jest.fn()
+
+const defaultPersistentDiskProps = {
+  diskExists: true,
+  computeConfig: defaultIComputeConfig,
+  updateComputeConfig,
+  handleLearnMoreAboutPersistentDisk: jest.fn(),
+}
+
+const defaultPersistentDiskTypeProps = {
+  diskExists: true,
+  computeConfig: defaultIComputeConfig,
+  updateComputeConfig,
+}
+
+jest.mock('src/libs/notifications', () => ({
+  notify: jest.fn((...args) => {
+    console.debug("######################### notify"); /* eslint-disable-line */
+    console.debug({ method: "notify", args: [...args] }); /* eslint-disable-line */
+  }),
+}))
+
+describe('compute-modal-component', () => {
+  describe('PersistentDiskType', () => {
+    // Passing diskExists [true,false] to PersistentDiskType
+    it('should be disabled when existing PD', async () => {
+      // Act
+      await render(h(PersistentDiskType, defaultPersistentDiskTypeProps))
```

Comment:
`render` does not return a Promise and thus does not need to be awaited. Similar comment throughout this file.

https://testing-library.com/docs/react-testing-library/api#render-result